GageStatsAgent/GageStatsAgent.cs

```diff
@@ -208,13 +208,26 @@ public IQueryable<String> GetRoles()
 
         #endregion 
         #region Station
-        public IQueryable<Station> GetStations()
+        public IQueryable<Station> GetStations(List<string> stationTypeList = null, List<string> agencyList = null, int page = 1)
```

Comment:
Thinking back to responsibilities/concerns of components related to pagination. I'm thinking that the services would be the responsibility of the controller to enforce pagination and not the agent. I would recommend moving the page information from the agent, such as order, skipping etc up to the controller level and out of the agent. The query to the database doesn't occur until after the request is processed or you enforce the request with a .ToList() or some equivalent request so you shouldn't have an issue with retrieval overhead